src/context/auth/authContext.js

```diff
@@ -0,0 +1,71 @@
+import { createContext, useContext, useReducer } from "react";
+import { loginApi, signupApi } from "../../services/auth_services";
+import { reducer } from "./authReducer";
+
+const AuthContext = createContext();
+const AuthContextProvider = ({ children }) => {
+  const user = JSON.parse(localStorage.getItem("user"));
+  const intialState = user ?? {
+    isUserloggedIn: false,
+    token: "",
+    user: null,
+  };
+console.log(user);
+  const [{ isUserloggedIn, token }, dispatch] = useReducer(
+    reducer,
+    intialState
+  );
+  const signup = async ({ email, password, firstName, lastName }) => {
+    const { data, success } = await signupApi({
+      firstName,
+      lastName,
+      email,
+      password
+    });
+    console.log(data); 
+    if (success) {
+      dispatch({
+        type: "UPDATE_USER",
+        payload: {
+          isUserloggedIn: true,
+          token: data.encodedToken,
+          user: data.createdUser,
+        },
+      });
+      console.log(data);
+      return true;
+    }
+    return false;
+  };
+
+  const login = async ({ email, password }) => {
+    const { data, success } = await loginApi({ email, password });
+    if (success) {
+      dispatch({
+        type: "UPDATE_USER",
+        payload: {
+          isUserloggedIn: true,
+          token: data.encodedToken,
+          user: data.foundUser,
+        },
+      });
+      console.log(data);
+      return true;
+    }
+    return false;
+  };
+  const logout = () =>{
+    dispatch({
```

Comment:
fix: In logout also remove the token stored in localstorage

Reply:
done already in reducer function .